File: lib/models/item.dart
import 'package:equatable/equatable.dart';
import 'package:cloud_firestore/cloud_firestore.dart';

class Item extends Equatable {
  final String iid;
  final String name;
  final String avatarURL;
  final List categoryIDs; // We add a category ID every time we construct a Competitor class

  const Item({
    required this.iid,
    required this.name,
    required this.avatarURL,
    required this.categoryIDs,
  });

  Map<String, dynamic> toMap() {
    return {
      'id': iid,
      'name': name,
      'avatarURL': avatarURL,
      'categoryIDs': categoryIDs,
    };
  }

  @override
  List<Object> get props {
    return [iid, name, avatarURL, categoryIDs];
  }

  @override
  bool get stringify => true;

  Item.fromMap(Map<String, dynamic> itemMap)
      : iid = itemMap["iid"],
        name = itemMap["name"],
        avatarURL = itemMap["avatarURL"],
        categoryIDs = itemMap["categoryIDs"];

  Item.fromDocumentSnapshot(DocumentSnapshot doc)
      : iid = doc.id,
        name = doc.get("name"),
        avatarURL = doc.get("avatarURL"),
        categoryIDs = doc.get("categoryIDs");
}

File: item.py

class Item():
    
    def __init__(self, name, avatarURL="", catIDs = None):
        self._name = name
        self._avatarURL = avatarURL
        if catIDs is not None:
            self._categoryIDs = catIDs
        else:
            self._categoryIDs = []

    def add_category(self, cid):
        self._categoryIDs.append(cid)

    def set_id(self, id):
        self._id = id

    def get_id(self):
        return self._id
    

    def fromDocumentSnapshot(doc):
        new = Item(doc.get("name"), doc.get("avatarURL"), doc.get("categoryIDs"))
        new.set_id(doc.id)
        return new

    def toMap(self):
        return {
            "iid": self._id,
            "name": self._name,
            "avatarURL": self._avatarURL,
            "categoryIDs": self._categoryIDs
        }